### PR TITLE
Support ORC format caching for iceberg connector

### PR DESCRIPTION
**Pull request by JySongWithZhangCe**
Fix https://github.com/prestodb/presto/issues/17261

```
== RELEASE NOTE ==

Iceberg Changes
* Support ORC format caching module for iceberg connector.
```

### CODE REVIEW

File: presto-iceberg/src/main/java/com/facebook/presto/iceberg/IcebergPageSourceProvider.java [hunk 1]
```diff
@@ -443,8 +442,12 @@ private static ConnectorPageSource createBatchOrcPageSource(
     {
         OrcDataSource orcDataSource = null;
         try {
-            FileSystem fileSystem = hdfsEnvironment.getFileSystem(user, path, configuration);
-            FSDataInputStream inputStream = hdfsEnvironment.doAs(user, () -> fileSystem.open(path));
+            ExtendedFileSystem fileSystem = hdfsEnvironment.getFileSystem(user, path, configuration);
```
Comment (rongrong) [inline presto-iceberg/src/main/java/com/facebook/presto/iceberg/IcebergPageSourceProvider.java:445]: While you are at it, let's clean it up a bit further. If you look at the call site of this function, you'll find that it calls `fileSystem.getFileStatus` to get file size then passes `fileSize` as a parameter to this function. Now that we are calling `getFileStatus` here for modification time as well, you can just remove the `getFileStatus` call from line 240-247.

File: presto-iceberg/src/main/java/com/facebook/presto/iceberg/IcebergPageSourceProvider.java [hunk 2]
```diff
@@ -443,8 +442,12 @@ private static ConnectorPageSource createBatchOrcPageSource(
...
+            FileStatus fileStatus = fileSystem.getFileStatus(path);
+            long modificationTime = fileStatus.getModificationTime();
+            HiveFileContext hiveFileContext = new HiveFileContext(true, NO_CACHE_CONSTRAINTS,
+                    Optional.empty(), Optional.of(fileSize), modificationTime, false);
+            FSDataInputStream inputStream = fileSystem.openFile(path, hiveFileContext);
```
Comment (rongrong) [inline presto-iceberg/src/main/java/com/facebook/presto/iceberg/IcebergPageSourceProvider.java:450]: I don't know the ins and outs of the APIs, but unless there's a good reason to not use `hdfsEnvironment.doAs`, let's keep the original behavior on this.

**Review by rongrong — COMMENTED** [review]
Thanks for working on this. Only some small improvements.

File: presto-iceberg/src/main/java/com/facebook/presto/iceberg/IcebergPageSourceProvider.java (hunk 1 above)
Comment (JySongWithZhangCe) [reply presto-iceberg/src/main/java/com/facebook/presto/iceberg/IcebergPageSourceProvider.java:434]: Done

File: presto-iceberg/src/main/java/com/facebook/presto/iceberg/IcebergPageSourceProvider.java (hunk 2 above)
Comment (JySongWithZhangCe) [reply presto-iceberg/src/main/java/com/facebook/presto/iceberg/IcebergPageSourceProvider.java]: Done

**Comment by JySongWithZhangCe** [comment]
> Please squash all commits into one and force push to update the branch.

Done

**Review by rongrong — APPROVED** [review]

**Comment by ChunxuTang** [comment]
Hi @JySongWithZhangCe, thanks for your contribution!
Just a very minor suggestion: Could you wrap your release note in the code markdown?
So it should be something like

```
== RELEASE NOTES ==

Iceberg Changes
* ...
```

instead of

== RELEASE NOTES ==
Iceberg Changes
* ...

**Comment by JySongWithZhangCe** [comment]
> Hi @JySongWithZhangCe, thanks for your contribution! Just a very minor suggestion: Could you wrap your release note in the code markdown? So it should be something like
> 
> ```
> == RELEASE NOTES ==
> 
> Iceberg Changes
> * ...
> ```
> 
> instead of
> 
> == RELEASE NOTES == Iceberg Changes
> 
> * ...

Done, thx @beinan

**Pull request merged**